### PR TITLE
Create proof of concept for integrating team filter

### PR DESCRIPTION
**Pull request by jhamet93**
Here's a rough implementation of filtering by a team. I wanted to get some feedback before refactoring and submitting.

1) What should happen when no team matches a filter ? Should an error be displayed and no games ? Should the whole schedule be shown ?
2) What should happen when you filter ? Shown a schedule with a possible games your filter matches ? Go automatically to the game if only one result returned ?

### CODE REVIEW

**Comment by xxhomey19** [comment]
1. I think it should show an error message and the whole schedule on that date.
2. Each team has only one game on the specific date for sure, so I think just go to the game directly. However, I believe some messages should be shown before going into live page, showing final box score or pregame comparison will be better.

Btw, user must enter optional date and filter in the meantime, right?
Oh! And don't forget to run linter before PR 👍

**Comment by fuloating** [comment]
Does it sound like doable to list games from the filtered team/player from more than one specific date? Like from - to, or "last week", "last month", "upcoming games" etc.?

**Comment by jhamet93** [comment]
@xxhomey19 

Date functionality works as before just with a layer of filtering on top. So it just filters all the games returned from a given date.
I implemented changes to output an error when no results are matched and then the entire schedule is returned.
Also, if one game is returned then that game will be chosen automatically.

@fuloating 

That would be a good enhancement. I just wanted to lay the groundwork done since I believe there's a lot of useful ideas for filtering such as a time range you specified or players.

**Comment by xxhomey19** [comment]
Cool! Whether this PR works in process or not?

**Comment by jhamet93** [comment]
@xxhomey19 

It works, check it out 😄 

`nba-go game --filter team=Detroit`

**Comment by xxhomey19** [comment]
@jhamet93 
Cool, It works!
However, I found that when I ran `npm run lint:fix` I would change some code format.
That is caused by prettier.  What's your prettier version? Mine is 1.7.4

Furthermore, add some description in README for this feature (GIF is not necessary). Thanks!

**Comment by jhamet93** [comment]
@xxhomey19 

I have 1.8.1 installed and prettier changes nothing when I lint it. Since the package version is prefixed with ^, it will update any major version so not sure if you want to lock the version down.

Just added some description to the read me for filtering game command.

**Comment by xxhomey19** [comment]
Got it. It's fine to use 1.8.1
Btw, I think you should rebase master, or other commits will be included in this PR.

**Comment by jhamet93** [comment]
@xxhomey19 

Cool, just did that. A couple files in this review I noticed were from fixing prettier changes on files that did not conform.

**Comment by xxhomey19** [comment]
@jhamet93 

Hey I wonder that is it possible the games could be filtered by team abbreviation?
Like
```
$ nba-go game --filter team=BOS
```
This should be better for user who don't want enter full team name.

**Comment by jhamet93** [comment]
@xxhomey19 

That actually is already implemented and works. A user can enter any abbreviation, mascot, city and as long as its specific enough, only that game will be returned.

Otherwise, all matching options (index of substring != -1) will be presented to select from.

File: README.md [hunk 1]
```diff
@@ -130,6 +130,16 @@ Check two teams' detailed scoreboard and box score.
...
+#### Filter
+
+Filter results to quickly jump to the info you care about
+
+#### Options
+##### `-f` or `--filter`
+
+Currently only supports filtering the results by team but more options on the way
+
+#### `nba-go game --filter team=Detroit`
```
Comment (xxhomey19) [inline README.md]:
```
$ nba-go game --filter team=Detroit
```
would be better.

**Comment by jhamet93** [comment]
@xxhomey19 

Good call, fixed that issue.

**Pull request merged**